src/MCPServer/lib/linkTaskManagerUnitVariableLinkPull.py

```diff
@@ -30,11 +30,23 @@
 
 class linkTaskManagerUnitVariableLinkPull(LinkTaskManager):
     def __init__(self, jobChainLink, pk, unit):
-        super(linkTaskManagerUnitVariableLinkPull, self).__init__(jobChainLink, pk, unit)
+        super(linkTaskManagerUnitVariableLinkPull, self).__init__(jobChainLink,
+                                                                  pk, unit)
+
+        # Look up the variable entry in the workflow data.
         var = TaskConfigUnitVariableLinkPull.objects.get(id=pk)
-        link = self.unit.getmicroServiceChainLink(var.variable, var.variablevalue, var.defaultmicroservicechainlink_id)
 
-        # Update the unit
-        if link is not None:
-            self.jobChainLink.setExitMessage(Job.STATUS_COMPLETED_SUCCESSFULLY)
-            self.jobChainLink.jobChain.nextChainLink(link, passVar=self.jobChainLink.passVar)
+        # Determine the next link.
+        link = self.unit.getmicroServiceChainLink(
+            var.variable,
+            var.variablevalue,
+            var.defaultmicroservicechainlink_id)
```

Comment:
This is an error I believe. `var.defaultmicroservicechainlink_id` here should be `var.defaultmicroservicechainlink`. Otherwise MCPServer triggers `AttributeError: 'unicode' object has no attribute 'id'` when the default is returned by `getmicroServiceChainLink()` and the `id` attr is accessed on a Unicode object.

This manifests as micro-service "Determine which files to identify" stalling/hanging.

Reply:
Another instance where mypy and type-checking would have caught this automatically.

Reply:
Ok sorry for the oversight, I think I probably missed this because I was only testing the code path used in that first magic link during transfer but not quarantine. I agree, mypy could be really helpful, unless you misuse union types haha.